File: js/tasks/new-year-chaos.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the minimumBribes function below.
function minimumBribes(q) {
    let bribes = 0;
    let moved = false;

    // more than 2 moves: Too chaotic
    for (let idx=0; idx< q.length; idx++) {
        let pos = q[idx];
        if ((pos - 1) - idx > 2)
            return 'Too chaotic';
    }

    // otherwise, execute bubble sort to count
    for (let i=0; i< q.length; i++) {
        for (let j=0; j< q.length; j++) {
            if (q[j] > q[j+1]) {
                let tmp = q[j]
                q[j] = q[j+1];
                q[j+1] = tmp;
                bribes++;
                moved = true;
            }
        }
        if (!moved) break;
        moved = false;
    }

    return bribes;
}

function main() {
    const t = parseInt(readLine(), 10);

    for (let tItr = 0; tItr < t; tItr++) {
        const n = parseInt(readLine(), 10);

        const q = readLine().split(' ').map(qTemp => parseInt(qTemp, 10));

        console.log(minimumBribes(q));
    }
}

File: python/tasks/new-year-chaos.py
#!/bin/python

import math
import os
import random
import re
import sys

# Complete the minimumBribes function below.
def minimumBribes(q):
    '''
    # comparing and insert/pop times out
    line = [ i+1 for i in xrange(len(q)) ]
    bribes = 0
    for x in xrange(len(q)):
        init = line.index(q[x])
        moves = init - x
        if moves > 2:
            return 'Too chaotic'
            return
        if moves > 0:
            bribes += moves
            line.insert(x, line.pop(init))
    '''

    # bubble sorting more efficient
    last = len(q) - 1
    bribes = 0
    moved = False

    # more than 2 moves: Too chaotic
    for idx, pos in enumerate(q):
        if (pos - 1) - idx > 2:
            return "Too chaotic"

    # otherwise, execute bubble sort to count
    for i in xrange(0, last):
        for j in xrange(0, last):
            if q[j] > q[j+1]:
                q[j], q[j+1] = q[j+1], q[j]
                bribes += 1
                moved = True
        if not moved:
            break 
        moved = False

    return bribes

if __name__ == '__main__':
    t = int(raw_input())

    for t_itr in xrange(t):
        n = int(raw_input())

        q = map(int, raw_input().rstrip().split())

        print minimumBribes(q)
